pkgs/applications/misc/squeekboard/default.nix

```diff
@@ -0,0 +1,91 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, cmake
+, meson
+, ninja
+, pkg-config
+, gnome3
+, glib
+, gtk3
+, wayland
+, wayland-protocols
+, rustc
+, cargo
+, libxml2
+, libxkbcommon
+, rustPlatform
+, makeWrapper
+, substituteAll
+, fetchpatch
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "squeekboard";
+  version = "1.9.1";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1nrihpffr91p3ipif5k5z9vcvkwxk6qy8hp93w13s7gnwl2g93hx";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    rustc
+    cargo
+    glib.dev
+    gtk3.dev
+    wayland
+    gnome3.gnome-desktop
+    makeWrapper
+  ];
+
+  buildInputs = [
+    glib.dev
+    gtk3.dev
+    gnome3.gnome-desktop
+    wayland
+    wayland-protocols
+    libxml2
+    libxkbcommon
+  ];
+
+  patches = [
```

Comment:
Also `patches` go by convention just after `src`, see https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887.